server/src/main/java/io/crate/auth/HttpAuthUpstreamHandler.java

```diff
@@ -110,7 +110,9 @@ private void handleHttpRequest(ChannelHandlerContext ctx, HttpRequest request) {
         }
 
         String username = credentials.username();
-        if (username != null && username.equals(authorizedUser)) {
+        if (username != null && username.equals(authorizedUser) && credentials.decodedToken() == null) {
+            // JWT authentication is stateless.
+            // For tokens with "exp" claim provided, it also ensures expiration check per-request.
```

Comment:
Honestly, I have no clue how these comments relates to this code.

Reply:
I rephrased and also moved the logic of "don't cache user" to another place
https://github.com/crate/crate/pull/15551/commits/081eb63cf13aa6709761995021f3ae6402785c9d

Reply:
I rephrased it